drivers/input/i8042prt/pnp.c

```diff
@@ -248,6 +248,41 @@ i8042DetectMouse(
 
     DeviceExtension->Flags |= MOUSE_PRESENT;
     INFO_(I8042PRT, "Mouse detected\n");
+	
+	/* Set the mouse to streaming mode, and disable error reporting (as per https://isdaman.com/alsos/hardware/mouse/ps2interface.htm) CORE-6901, CORE-12663, CORE-12434 */
+	i8042IsrWritePort(DeviceExtension, MOU_SET_STREAM_MODE, CTRL_WRITE_MOUSE);
+	i8042IsrWritePort(DeviceExtension, MOU_NO_ERROR_REPORT, CTRL_WRITE_MOUSE);
+	
+	/* Initialize Intellimouse or Intellimouse Explorer compatible mice (This fixes wheels) */
+	i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+	i8042IsrWritePort(DeviceExtension, 0xC8, CTRL_WRITE_MOUSE);
+
+	i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+	i8042IsrWritePort(DeviceExtension, 0x64, CTRL_WRITE_MOUSE);
+	
+	i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+	i8042IsrWritePort(DeviceExtension, 0x50, CTRL_WRITE_MOUSE);
+	
+	i8042IsrWritePort(DeviceExtension, MOU_READ_DEV_TYPE, CTRL_WRITE_MOUSE);
+	i8042ReadDataWait(DeviceExtension, &Value);
+	//TODO: Help me! The value of MouseType doesn't actually get updated so the wheel logic never gets appplied 
+	if(Value == 0x03)
+	{
+		
+        IN I8042_MOUSE_EXTENSION DeviceExtension;
```

Comment:
Wth is that "IN" thing, and why is this variable declared there and not used afterwards?